### DISCUSSION TITLE
What's the difference between packages/tools and apps/libs directory structures?

### DISCUSSION THREAD

**Discussion by amartincolby** [nrwl/nx#18035] [General]
Exactly as the title asks: what's the difference between packages/tools and apps/libs directory structures? I have found a few vague mentions of a difference but no deep elaboration beyond that. If I scaffold a TS/JS monorepo, I get `packages` and `tools` directories, but if I scaffold a React application, I see `apps` and `libs`. Why? Is this an example of duplicated ways to solve the same thing? UPDATE: After re-reading, it seems like I scaffolded a standalone React application. I did not. I created an integrated monorepo in both cases.

**Comment by withkynam** [comment]
hey @amartincolby , have you managed to figure it out? I have the same questions about the difference between the app and package folder. Will those store different projects and tools/libs store the different utilities that might be used in any project?

**Reply by amartincolby** [reply]
Nope. No official response. After working with a number of Nx instances, it appears to be a pure naming difference. I suspect that the difference comes from disparate origins of some of Nx's structure, i.e. Angular vs React. It was never a problem but it continues to annoy the hell out of me.

**Comment by edbzn** [comment]
Hi, there's no fundamental difference between the two layouts. It comes from the fact that Nx can be used for **integrated** monorepos where the convention is to use the "apps/libs" layout but also for **package-based** repos where the convention is to use the "packages" layout, the latter one is traditionally used for publishing libraries and the former one for developing apps.

Still, Nx lets you configure the directory structure, it's up to you to decide which layout to choose, the "packages" layout can be used for integrated monorepos and vice-versa. Note that from my perspective, even if "apps/libs" is the default layout for developing apps, the "packages" layout can be beneficial for some reasons: 

- code colocation (no need to navigate between two different folders)
- code owner configuration (easier to configure)
